cmd/upgrade/roles/cmd.go

```diff
@@ -1173,3 +1180,16 @@ func checkPolicyAndClusterVersionCompatibility(policyVersion, clusterVersion str
 	}
 	return nil
 }
+
+func printClusterUpgradeInfo(mode string, reporter *rprtr.Object) {
```

Comment:
@mikeshng I'd say lets be consistent in our usage of global versus function parameters and avoid globals when we can. So how about:

```
printClusterUpgradeInfo(clusterKey string, clusterUpgradeVersion string, mode string, reporter *rprtr.Object)
```

as a function signature?
```

and remove the `if args.isInvokedFromClusterUpgrade` expression from the function as you're already guarding the function invocation with that anyway in the command.

It makes the function easier to test if we're not reliant on global state.

Reply:
You could equally return the command as a `string` from the function, drop the `runtime` parameter and write a unit test for this :-)

So your caller then becomes:

```
if args.isInvokedFromClusterUpgrade {
	reporter.Infof(printClusterUpgradeInfo(args.clusterId, args.clusterUpgradeVersion, mode)
}
```

Reply:
Done as suggested. Thanks.

Reply:
Done as suggested after making some fixes :) Sorry for the email spams.